executor/analyze.go

```diff
@@ -897,8 +904,34 @@ func (e *AnalyzeFastExec) handleSampTasks(bo *tikv.Backoffer, workID int, err *e
 }
 
 func (e *AnalyzeFastExec) buildHist(ID int64, collector *statistics.SampleCollector, tp *types.FieldType) (*statistics.Histogram, error) {
-	// TODO: build histogram and cmsketch here for one collector.
-	return nil, nil
+	// build collector properties.
+	collector.Samples = collector.Samples[:e.sampCursor]
+	err := collector.CalcTotalSize()
+	if err != nil {
+		return nil, err
+	}
+	collector.Count = int64(e.sampCursor)
+	data := make([][]byte, 0, len(collector.Samples))
+	for _, sample := range collector.Samples {
+		if sample.Value.IsNull() {
+			collector.NullCount++
+			continue
+		}
+		bytes, err := sample.Value.ToBytes()
+		if err != nil {
+			return nil, err
+		}
+		data = append(data, bytes)
+	}
+	rowCount := mathutil.MinInt64(domain.GetDomain(e.ctx).StatsHandle().GetTableStats(e.tblInfo).Count, int64(e.rowCount))
```

Comment:
Does `e.rowCount` count MVCC older version rows?

Reply:
Yeah. It contains.

Reply:
So if numbers of versions for rows are high, it may impact the randomness of the fast sampling?

Reply:
Yeah. But we can't avoid this situation.